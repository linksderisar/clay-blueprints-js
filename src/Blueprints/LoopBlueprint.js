import Blueprint from './Blueprint';

export default class LoopBlueprint extends Blueprint {
    constructor (iterable) {
        super();
        this._iterable = iterable;
    }

    get iterable() {
        return this._iterable;
    }

    set iterable(value) {
        this._iterable = value;
    }

    toObject() {
        return {
            'loop': this.iterable
        };
    }
}
