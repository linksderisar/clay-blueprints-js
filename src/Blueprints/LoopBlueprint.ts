import AbstractBlueprint from './Abstracts/AbstractBlueprint';

export default class LoopBlueprint extends AbstractBlueprint {

    iterable: string = '';

    public getIterable(): string {
        return this.iterable;
    }

    public setIterable(iterable: string): this {
        this.iterable = iterable;
        return this;
    }

    toObject(): any {
        return {
            loop: this.iterable,
        };
    }
}
