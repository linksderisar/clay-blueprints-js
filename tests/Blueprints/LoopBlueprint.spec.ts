import LoopBlueprint from '../../src/Blueprints/LoopBlueprint';

it('can be converted', () => {
    const blueprint = LoopBlueprint.create('loop.ref');

    expect(blueprint.toObject()).toMatchSnapshot();
});
