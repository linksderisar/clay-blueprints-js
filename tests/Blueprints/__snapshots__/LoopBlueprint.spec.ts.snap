// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can be converted 1`] = `
Object {
  "loop": "",
}
`;
